### PR TITLE
refined the NFS Ganesha cluster procedure

### PR DESCRIPTION
**Pull request by klgill**
https://issues.redhat.com/browse/RHOSPDOC-2031

### CODE REVIEW

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:5]: Should this be CephFS via NFS?

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:10]: Is there a command to identify the nodes? Or are we just saying that the customer needs to decide which nodes they want to deploy the new NFS service on?

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:18]: Is this whole procedure relevant for Ceph Storage nodes that were deployed via director? Or only these steps, and then the rest of the procedure is relevant to both director-deployed Ceph and external Ceph?

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:11]: Is it correct to say, "...identify the node definition file that is used in the **RHOSP** environment"?

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:18]: And do we need to be any more specific about how to identify the node definition file?

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:76]: I don’t see { VIP}. I see “Fixed IP Addresses”.

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:97]: Is it okay to link to the Red Hat Ceph Storage 7 Operations Guide downstream? https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/7/html-single/operations_guide/index#management-of-nfs-ganesha-gateway-using-the-ceph-orchestrator

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:88]: I’m confused by the placement of this list. The information about picking an appropriate size of the cluster and setting the ingress-mode sounds like it belongs in the intro or prerequisites.

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [inline docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:100]: Why is it relevant to mention that the “following commands are run inside the cephadm shell” here? Should these commands be steps?

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:76]: yeah, this is Because the VIP is retrieved by the `openstack port list ..` command, that filters the output on the "Fixed IP Addresses" column, where we can find the VIP assigned by TripleO to the StorageNFS network.

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:100]: I agree the following content should be rewritten under form of steps (I assume it makes the style consistent with the rest of the documentation).
In the snippet below, you can use the comments to build the statement related to the step, and then put the command associated.

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:97]: Yes.
downstream we link the RHCS 7 ops guide, while upstream we point to docs.ceph.com.

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:88]: I think we should remove `{{ cluster_size }}` and decide a size in advance. In theory this is driven by the placement "by label" (nfs is the label here), so we do not need to specify the cluster size.
I propose to remove the following:
```
* Pick an appropriate size for the NFS cluster. The NFS service provides
active/active high availability when the cluster size is more than
one node. It is recommended that the ``{{ cluster_size }}`` is at least one
less than the number of hosts identified. This solution has been tested
with a 3-node NFS cluster.
```

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:100]: Also remove "{{ cluster size }}" from the `ceph nfs cluster create ` command.

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:18]: @fmount I'm not sure if you saw these 2 comments above. ^

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:10]: @fmount 
Maybe something like the following?
[NOTE]
You can deploy the new clustered NFS service on your existing CephStorage nodes or HCI nodes, or on new hardware that you enrolled in the Ceph cluster.

Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:10]: +1 to have a note, thank you!

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:18]: Sorry, I missed this comment (the GH interface doesn't help me when there are a lot of comments). Replying ....

Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:18]: Either if the Ceph is director deployed or external, we used to deploy Ganesha with TripleO. Now this pattern changes in 18, and we ask, as part of this guide, to redeploy ganesha (aka ceph-nfs cluster) in the nodes that are part of Ceph, and that are not going to be decommissioned.
This means that 90 % of this procedure applies to both use cases, except for this part related to TripleO, that doesn't apply to a cluster that is externally managed.
However, we might want to put a note about external Ceph, saying that in that case, the human administrator should take care of:
1. choosing the network where the ceph-nfs should be exposed (in terms of VIP/frontend network)
2. select the VIP that will be used for Ganesha when the `ceph nfs create ...` command is run.

@klgill do we want to make this distinction here, at the very beginning of the procedure, and put a Note related to the external Ceph cluster scenario?

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:11]: No, they usually don't change the name, which is why I would refer to the existing guide to help users identify this file, that in general has that name and there's no reason to change it.

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:76]: Yes, that makes sense.

File: docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc
Comment (klgill) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:18]: @fmount OK, will need to change step 2 back to its original format, which was that the director-deployed steps were substeps under step 2. And I can write "If you are using director-deployed Ceph, propogate the StorageNFS network..." Then maybe have a note about external Ceph below step 2.

Who is doing the director-deployed steps, if not a human administrator?

Comment (fmount) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc:25]: ok thanks, and yes, a human administrator in both cases, it's just that it might be not the same person when Ceph is external (Ceph cluster admin != openstack admin), but this is a minor thing and up to the customer, it's human in any case :D

Comment (klgill) [reply docs_user/modules/proc_creating-a-ceph-nfs-cluster.adoc]: @fmount I formatted step 2 as substeps again, and introduced the step with, "If you deployed your Red Hat Ceph Storage nodes with director..."

Can you also review steps 4-7? Those are the commands that I converted to steps. 
After I get your approval, I'll send this PR to peer review.

**Review by fmount — APPROVED** [review]
/lgtm

**Pull request merged**
